File: sdk/.net/ardvro.sdk.net/component/sqljson/controller/CommonController.cs
namespace ardvro.sdk.net.component.sqljson.controller;


using ardvro.core.ext.connection;
using ardvro.core.lib.common;

public class CommonController
{
    private IConnection _conection;

    public CommonController(IConnection connection)
    {
        _conection = connection;
    }

    public void GenerateRsaKey(Action<Flexible> onResponse)
    {
        String functionName = "ardvro/component/sqljson/Common/GenerateRsaKey";
        _conection.Submit(new Object[]{ }, functionName, onResponse);
    }

    public void GenerateRsaServerKey(Action<String> onResponse)
    {
        String functionName = "ardvro/component/sqljson/Common/GenerateRsaServerKey";
        _conection.Submit(new Object[]{ }, functionName, onResponse);
    }

    public void GenerateRsaClientKey(Action<String> onResponse)
    {
        String functionName = "ardvro/component/sqljson/Common/GenerateRsaClientKey";
        _conection.Submit(new Object[]{ }, functionName, onResponse);
    }

}

File: sdk/web/component/sqljson/controller/CommonController.js
var CommonController = function CommonController(cfg)
{
    let ctrl = {};

    ctrl.GenerateRsaKey = function (callback)
    {
        cfg.Connector.Submit([], "ardvro/component/sqljson/Common/GenerateRsaKey", function (box)
        {
            callback != null ? callback(box == null ? null : box.Data) : null;
        });
    };

    ctrl.GenerateRsaServerKey = function (callback)
    {
        cfg.Connector.Submit([], "ardvro/component/sqljson/Common/GenerateRsaServerKey", function (box)
        {
            callback != null ? callback(box == null ? null : box.Data) : null;
        });
    };

    ctrl.GenerateRsaClientKey = function (callback)
    {
        cfg.Connector.Submit([], "ardvro/component/sqljson/Common/GenerateRsaClientKey", function (box)
        {
            callback != null ? callback(box == null ? null : box.Data) : null;
        });
    };


    return ctrl;
};